testsuite/tests/templates/basic/bad_instantiate_not_arg.reference

```diff
@@ -0,0 +1,2 @@
+File "_none_", line 1:
+Error: Argument must be compiled with -as-argument-for: Widget
```

Comment:
> Error: Module `Widget` cannot be used as argument for instantiation of `List_monoid`.
> Hint: Compile `Widget.cmi` with `-as-argument-for Foo` where `Foo` is a parameter of `List_monoid`.
> Hint: parameters of `List_monoid`: XXX

Reply:
Leaving out the "for instantiation of" bit since it makes it sound like it's not compatible with `List_monoid` in particular, when actually it can't be used as an argument for instantiating _anything_

Reply:
Good point. Then maybe:
> Error: Module Widget cannot be used as argument because it was not compiled as such.
Hint: Compile Widget.cmi with -as-argument-for followed by one of parameters of List_monoid: XXX, YYY, ZZZ

Reply:
Oh, also, listing out the parameters here is a good idea but it turns out to be very awkward; I'll add a CR